src/components/custom/like/like-heart.tsx

```diff
@@ -0,0 +1,61 @@
+"use client"
+
+import fetchDislikeNft from "@/lib/query/nfts/fetchDislikeNft"
+import fetchLikeNft from "@/lib/query/nfts/fetchLikeNft"
+import type { NftFeedType } from "@/validators/schemas/nfts/feedSchema"
+import { useMutation } from "@tanstack/react-query"
+import { useTranslations } from "next-intl"
+import { useCallback, useState } from "react"
+import { FaHeart, FaRegHeart } from "react-icons/fa"
+import { toast } from "sonner"
+
+export default function LikeHeart(props: LikeHeartProps) {
+  const t = useTranslations("like")
+  const { nft, isLiked, iconSize } = props
+  const [liked, setLiked] = useState(isLiked)
```

Comment:
If you needed it as function because of the `useState`, then better create the function from the props here, rather than in the parent.